registry_no_code_module.go

```diff
@@ -41,6 +45,44 @@ type RegistryNoCodeModules interface {
 	UpgradeWorkspace(ctx context.Context, noCodeModuleID string, workspaceID string, options *RegistryNoCodeModuleUpgradeWorkspaceOptions) (*WorkspaceUpgrade, error)
 }
 
+// RegistryModuleVariableList is a list of registry module variables.
+type RegistryModuleVariableList struct {
+	Items []*RegistryModuleVariable
+
+	// NOTE: At the time of authoring this comment, the API endpoint to fetch
+	// registry module variables does not support pagination. This field is
+	// included to satisfy jsonapi unmarshaler implementation here:
+	// https://github.com/hashicorp/go-tfe/blob/3d29602707fa4b10469d1a02685644bd159d3ccc/tfe.go#L859
```

Comment:
I think this could be a bug in the unmarshaler implementation. In the case of the change in this PR, `Items` is a slice, and in the `unmarshalResponse` function, it should be handled by `jsonapi.UnmarshalManyPayload`. However, without the `*Pagination` field of this struct present (which should not actually be here, because the backend service doesn't actually return a `*Pagination) in the response), the conditional [here](ttps://github.com/hashicorp/go-tfe/blob/3d29602707fa4b10469d1a02685644bd159d3ccc/tfe.go#L859) sends the request into the `jsonapi` handler `UnmarshalPayload`, which is documented to support "single request payloads only", leading an attempt to decode a slice to fail.

For now, just including the `*Pagination` field here is a workaround. But if this issue is fixed, then it can be removed (or if pagination is implemented on the backend, it should be kept).

Reply:
This is a great callout, I'll create a ticket to look into this. :+1: